src/hooks/use-handler/use-handler.js

```diff
@@ -0,0 +1,158 @@
+/** @format */
+
+import { useNavigate } from "react-router-dom";
+import { useAuth, useToast, useVideos } from "../../context";
+import axios from "axios";
+import { useState } from "react";
+
+export const useHandler = () => {
+  const { videoDispatch } = useVideos();
+  const {
+    authState: { token },
+  } = useAuth();
+
+  const { setToast } = useToast();
+  const navigate = useNavigate();
+  const [loading, setLoading] = useState(false);
+
+  // this function handle all the api calls
+
+  const serverCalls = async (
+    method,
+    url,
+    type,
+    property,
+    message,
+    body = null
+  ) => {
+    const headers = { authorization: token };
+    try {
+      setLoading(true);
+      const { data } = await axios({
+        method: method,
+        url: url,
+        data: body,
+        headers: headers,
+      });
+      console.log("data coming from server", data);
+      videoDispatch({ type, payload: data[property] });
+      setToast({
+        toastVarient: "success",
+        message: message,
+        toast: true,
+      });
+      setLoading(false);
+    } catch (error) {
+      setToast({
+        toast: true,
+        toastVarient: "failure",
+        message: "Something went wrong!!",
```

Comment:
instead of writing hardcoded error messages, you can extract error messages using "error.response.data.errors"